services/device/geolocation/geoclue_location_provider.h

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_SERVICES_DEVICE_GEOLOCATION_GEOCLUE_LOCATION_PROVIDER_H_
+#define BRAVE_SERVICES_DEVICE_GEOLOCATION_GEOCLUE_LOCATION_PROVIDER_H_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/memory/scoped_refptr.h"
+#include "base/memory/weak_ptr.h"
+#include "base/sequence_checker.h"
+#include "dbus/bus.h"
+#include "dbus/message.h"
+#include "dbus/object_path.h"
+#include "dbus/object_proxy.h"
+#include "dbus/property.h"
+#include "services/device/public/cpp/geolocation/location_provider.h"
+#include "services/device/public/mojom/geoposition.mojom.h"
+
+namespace device {
+
+struct GeoClueLocationProperties;
+
+class GeoClueLocationProvider : public LocationProvider {
```

Comment:
`LocationProviderLinux` or `GeoClueLocationProviderLinux`

Reply:
`LocationProviderLinux` or `GeoClueLocationProviderLinux` would match `LocationProviderAndroid` and `LocationProviderWinRT`. The others are more generic like `NetworkLocationProvider`